Library/Homebrew/test/cask/artifact/app_spec.rb

```diff
@@ -334,5 +334,29 @@
 
       expect(contents_path).to exist
     end
+
+    it "properly handles non-writable directories" do
+      install_phase
+
+      contents_path = target_path.join("Contents")
+
+      expect(app.target).to receive(:writable?).at_least(:once).and_return false
+      allow(command).to receive(:run!).and_call_original
+      allow(command).to receive(:run!)
```

Comment:
I tried that, but it doesn't work because the test system assumes this particular `expect` is the only call made and fails because the upgrade step runs other commands as well.

Reply:
I was able to work around it by adding another `expect` call with `any_args`.